src/plugins/analysis/hardware_analysis/code/hardware_analysis.py

```diff
@@ -0,0 +1,88 @@
+from typing import Optional
+
+from analysis.PluginBase import AnalysisBasePlugin
+
+
+class AnalysisPlugin(AnalysisBasePlugin):
+    '''
+    Hardware Analysis Plug-in
+    '''
+    NAME = 'hardware_analysis'
+    DESCRIPTION = 'Hardware Analysis Plug-in'
+    DEPENDENCIES = ['cpu_architecture', 'elf_analysis', 'kernel_config']
+    VERSION = '0.2'
+
+    def __init__(self, plugin_adminstrator, config=None, recursive=True):
+
+        self.config = config
+
+        super().__init__(plugin_adminstrator, config=config, recursive=recursive, plugin_path=__file__)
+
+    def process_object(self, file_object):
+
+        # search for important information
+        cpu_architecture = self.cpu_architecture_analysis(file_object)
+        modinfo = self.filter_modinfo(file_object)
+        kernel_config = self.filter_kernel_config(file_object)
+
+        # store the results
+        file_object.processed_analysis[self.NAME] = dict()
+        file_object.processed_analysis[self.NAME]['cpu architecture'] = cpu_architecture
+        file_object.processed_analysis[self.NAME]['*.ko ".modinfo"'] = modinfo
+        file_object.processed_analysis[self.NAME]['kernel configuration'] = kernel_config
```

Comment:
We can clear this up a bit:
```suggestion
        file_object.processed_analysis[self.NAME] = {
            'cpu architecture': cpu_architecture,
            'modinfo section': modinfo,
            'kernel configuration': kernel_config
        }
```
Also `'*.ko ".modinfo"'` doesn't seem to be very practical for accessing the data so maybe something like "modinfo section" would be better